pilot/pkg/security/model/authentication.go

```diff
@@ -188,7 +180,7 @@ func ConstructSdsSecretConfig(name string) *tls.SdsSecretConfig {
 			// set the fetch timeout to 0 here because workload certs are
 			// guaranteed to exist. while others like gateway certs may not
 			// exist.
-			InitialFetchTimeout: fetchTimeout,
+			InitialFetchTimeout: ptypes.DurationProto(time.Second * 0),
```

Comment:
why do you change the fetchTimeout here? This one should remain the old one. As you will only change the InitialFetchTimeout to 0 when the name is SDSRootResourceName or SDSDefaultResourceName

Reply:
@adiprerepa Sorry just find that you change here could you fix this logic?

Reply:
oh, so only set to 0 when the name is `SDSRootResourceName` or `SDSDefaultResourceName`? got it

Reply:
yes no need to change it here. as it is only for rootca and default case you will set the InitialFetchTimeout to 0. change it back I think will be good. and all other parts are good to me thanks